### ISSUE TITLE
Battle softlocks and minor bugs

### ISSUE THREAD

**Issue by BobLocke** [abyssonym/smrpg_gbarp#9]
Hey, really enjoying this. I've run into a few softlocks you might want to know about.

Seed: 1474862724
- Mario using Group Hug during the Axem Rangers fight caused a softlock. Hasn't happened in any other battles that I've observed. Bowser could use Group Hug in the same fight with no problems.
- Using an item on a "Drill Bit" when it is the only enemy left in battle causes a soft lock. It appears they can't be targeted by items.

Lastly, I don't know if this is something you can fix, but when Yoshi Cookies get you an equipment piece from an enemy, it goes to the wrong inventory and can only be sold or "used" for no effect. Is it possible to get these items to go to the equipment inventory?

**Comment by BobLocke** [comment]
Just ran into another one:
- The chandeliers in the Boomer fight are targetable with attacks, and if you kill them the post-boss cutscene hangs indefinitely.

**Comment by abyssonym** [comment]
- [ ] Group Hug - I'm not able to reproduce this bug, so I'm not sure how to fix it.
- [x] Drill Bit - I think this is not related to Drill Bit specifically. I encountered a similar issue on battles with more than 6 enemies. Drill Bit is one of the enemies that can have up to 8 in one battle, so I suspect it might be the same issue. Let me know if I'm wrong or if the issue persists in version 3.
- [x] Yoshi Cookies - Making the items go to the appropriate menu would probably require ASM hacking. Yoshi Cookies will only turn enemies into consumable items, for the time being.
- [x] Not sure what's going on here but I've added the Hangin' Shys to the boss pool, so they won't be randomized in weird ways.
